app/scripts/lib/createRPCMethodTrackingMiddleware.js

```diff
@@ -160,8 +162,37 @@ export default function createRPCMethodTrackingMiddleware({
 
       const properties = {};
 
+      let msgParams;
+
       if (event === EVENT_NAMES.SIGNATURE_REQUESTED) {
         properties.signature_type = method;
+
+        const data = req?.params?.[0];
+        const from = req?.params?.[1];
+        const paramsExamplePassword = req?.params?.[2];
+
+        msgParams = {
+          ...paramsExamplePassword,
+          from,
+          data,
+          origin,
+        };
+
+        const msgData = {
+          msgParams,
+          status: 'unapproved',
+          type: req.method,
+        };
+
+        const securityProviderResponse = await securityProviderRequest(
+          msgData,
+          req.method,
+        );
```

Comment:
`securityProviderRequest` may throw an error from metamask-controller.js which could break the flow here. I think we should catch errors here to not allow errors from `securityProviderCheck` to stop the dapp request. 

Current error flow that may occur:
- we call `securityProviderRequest` here in createRPCMethodTrackingMiddleware.js
- `securityProviderRequest` (metamask-controller.js)  calls `securityProviderCheck` to fetch validation
- error occurs when calling `securityProviderCheck`
- we catch the error in metamask-controller.js and then also throw the error again
- error breaks createRPCMethodTrackingMiddleware.js and stops dapp request

Reply:
hey @filipsekulic, I went ahead and merged a solution for the above concern because it seems we're aiming for v10.27 and I was pretty close to a solution after debugging/reviewing. does this look ok to you? [4288143](https://github.com/MetaMask/metamask-extension/pull/17688/commits/428814312a7870ed6c5ede61e4f45370ab28b753)

Reply:
cc: @brad-decker @jpuri

Reply:
It seems fine to me